posts/2021-06-17-Rust-1.53.0.md

```diff
@@ -0,0 +1,180 @@
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.53.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.53.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1530-2021-06-17
+
+## What's in 1.53.0 stable
+
+This release contains several new language features and many new library features,
+including the long-awaited `IntoIterator` implementation for arrays.
+See the [detailed release notes](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1530-2021-06-17)
+to learn about other changes not covered by this post.
+
+### IntoIterator for arrays
+
+This is the first Rust release in which arrays implement the `IntoIterator` trait.
+This means you can now iterate over arrays by value:
+
+```rust
+for i in [1, 2, 3] {
+    ..
+}
+```
+
+Previously, this was only possible by reference, using `&[1, 2, 3]` or `[1, 2, 3].iter()`.
+
+Similarly, you can now pass arrays to methods expecting a `T: IntoIterator`:
+
+```rust
+let set = BTreeSet::from_iter([1, 2, 3]);
+```
+
+```rust
+for (a, b) in some_iterator.chain([1]).zip([1, 2, 3]) {
+    ..
+}
+```
+
+This was not implemented before, due to backwards compatibility problems.
+Because `IntoIterator` was already implemented for references to arrays,
+`array.into_iter()` already compiled in earlier versions,
+resolving to `(&array).into_iter()`.
+
+As of this release, arrays implement `IntoIterator` with a small workaround to avoid breaking code.
+The compiler will continue to resolve `array.into_iter()` to `(&array).into_iter()`,
+as if the trait implementation does not exist.
+This only applies to the `.into_iter()` method call syntax, and does not
+affect any other syntax such as `for e in [1, 2, 3]`, `iter.zip([1, 2, 3])` or
+`IntoIterator::into_iter([1, 2, 3])`, which all compile fine.
+
+Since this special case for `.into_iter()` is only required to avoid breaking existing code,
+it is removed in the new edition, Rust 2021, which will be released later this year.
+See [the edition announcement](https://blog.rust-lang.org/2021/05/11/edition-2021.html#intoiterator-for-arrays)
+for more information.
+
+### Or patterns
+
+Pattern syntax has been extended to support `|` nested anywhere in the pattern.
+This enables you to write `Some(1 | 2)` instead of `Some(1) | Some(2)`.
+
+```rust
+match result {
+     Ok(Some(1 | 2)) => { .. }
+     Err(MyError { kind: FileNotFound | PermissionDenied, .. }) => { .. }
+     _ => { .. }
+}
+```
+
+### Unicode identifiers
+
+Identifiers can now contain non-ascii characters.
+All valid identifier characters in Unicode as defined in [UAX #31](https://unicode.org/reports/tr31/) can now be used.
```

Comment:
Since no layperson is expected to be familiar with UAX 31, and since the number one question on people's minds is going to be "does this support emoji?", it's probably worth expanding here that this is only for supporting characters that get used in identifiers in natural languages, and doesn't support emoji, box-drawing characters, etc.

Reply:
Ah, good idea.

```
error: unknown start of token: \u{1f988}
 --> src/main.rs:2:9
  |
2 |     let 🦈 = 4;
  |         ^^
```
:(